src/Client/V2/Access/DatabaseAccess.cs

```diff
@@ -389,11 +411,211 @@ public virtual DatabaseQueryResult Query( string databaseId, DatabaseQuery query
+        /// </param>
+        /// <param name="context">
+        /// The optional call context to include.
+        /// </param>
+        public virtual async IAsyncEnumerable<IReadOnlyDictionary<string, object>> QueryDictionaryAsync( string databaseId, DatabaseQuery query, int rowsPerCall = DefaultRowsPerPage, CallContext context = null )
+        {
+            AsyncQueryInfo info = null;
+            try
+            {
+                info = await StartQueryAsync( databaseId, query, context );
+                string[] orderedIds = GetOrderedFieldIds( info );
+
+                int startingRow = 0;
+                bool moreToRead = true;
+                while( moreToRead )
+                {
+                    AsyncQueryData data = await ReadQueryAsync( databaseId, info.Id, startingRow, startingRow + rowsPerCall - 1, context );
+                    foreach( ICollection<object> row in data.Rows )
+                    {
+                        int i = 0;
+                        var result = new Dictionary<string, object>();
+                        foreach( object value in row )
+                        {
+                            result.Add( orderedIds[i], ScrubDatabaseValue( value ) );
+                            i++;
+                        }
+
+                        yield return result;
+                    }
+                    startingRow += rowsPerCall;
+                    moreToRead = data.HasMoreRows;
+                }
+            }
+            finally
+            {
+                if( info != null )
+                    await StopQueryAsync( databaseId, info.Id, context );
+            }
+        }
+
+        /// <summary>
+        /// Reads all of the values for an async database query and returns them as an async enumerable of dictionaries.
+        /// The keys in the returned dictionary are the field ids, and the values are the values for each field represented
+        /// by the field id. This will throw an <see cref="System.ArgumentException"/> if a field is specified more than
+        /// once in the query due to attempting to add duplicate dictionary keys and QueryValuesAsync should be used instead.
+        /// </summary>
+        /// <param name="databaseId">
+        /// The unique identifier of the EMS database being queried.
+        /// </param>
+        /// <param name="info">
+        /// The asynchronous query info, as retrieved from <see cref="StartQueryAsync(string, DatabaseQuery, CallContext)"/>.
+        /// </param>
+        /// <param name="rowsPerCall">
+        /// The number of rows to read for each API call.
+        /// </param>
+        /// <param name="context">
+        /// The optional call context to include.
+        /// </param>
+        public virtual async IAsyncEnumerable<IReadOnlyDictionary<string, object>> ReadQueryDictionaryAsync( string databaseId, AsyncQueryInfo info, int rowsPerCall = DefaultRowsPerPage, CallContext context = null )
```

Comment:
Ah... is there a reason why `QueryDictionaryAsync` can't just call this in its body?

Reply:
This is weird IAsyncEnumerable shit, but basically I would have to call it with an `await foreach` in order to use it here, and maybe that is very little overhead but I'm not sure?